tools/python-frame-reference.py

```
@@ -231,57 +253,135 @@ def _build_default_path_vector() -> tuple[dict, str]:
+    # before the payload: magic, version, header length, header bytes.
+    legacy_prefix = legacy["frame_hex"][: len(legacy["frame_hex"]) - len(legacy["expected_payload_hex"])]
+    twin_prefix = twin["frame_hex"][: len(twin["frame_hex"]) - len(twin["expected_payload_hex"])]
+    _require(twin_prefix == legacy_prefix, "twin frame prefix (magic/version/header bytes) differs from the legacy vector")
+    for field in ("compressed_data_hex", "checksum_hex", "original_size", "format", "inner_msgpack_hex"):
+        _require(
+            twin["payload_envelope"][field] == legacy["payload_envelope"][field],
+            f"twin payload_envelope.{field} differs from the legacy vector — encoding must be the ONLY delta",
+        )
 
 
 def generate() -> int:
```

Ruff (0.16.2)

[warning] Too many statements (56 > 50)

(PLR0915)